cmd/clusterctl/phases/pivot.go

```diff
@@ -271,16 +334,24 @@ func moveMachineSets(from sourceClient, to targetClient, machineSets []*clusterv
 
 func moveMachineSet(from sourceClient, to targetClient, ms *clusterv1.MachineSet) error {
 	klog.V(4).Infof("Moving MachineSet %s/%s", ms.Namespace, ms.Name)
+
 	klog.V(4).Infof("Retrieving list of Machines for MachineSet %s/%s", ms.Namespace, ms.Name)
 	machines, err := from.GetMachinesForMachineSet(ms)
 	if err != nil {
 		return err
 	}
-
 	if err := moveMachines(from, to, machines); err != nil {
 		return err
 	}
 
+	// Move infrastructure reference if the MachineSet isn't linked to a MachineDeployment.
```

Comment:
Why?

Reply:
If it's in a MachineDeployment, we already have moved the referenced template

Reply:
Let's add a comment to that effect, pls.